articles/quickstart/native/android-vnext/00-login.md

```diff
@@ -0,0 +1,238 @@
+:::
+
+## Add Login to your App
+
+[Universal Login](/hosted-pages/login) is the easiest way to set up authentication in your application. We recommend using it for the best experience, best security and the fullest array of features.
+
+::: note
+You can also embed the login dialog directly in your application using the [Lock widget](/lock). If you use this method, some features, such as single sign-on, will not be accessible.
+To learn how to embed the Lock widget in your application, follow the [Embedded Login sample](https://github.com/auth0-samples/auth0-android-sample/tree/embedded-login/01-Embedded-Login). Make sure you read the [Browser-Based vs. Native Login Flows on Mobile Devices](/tutorials/browser-based-vs-native-experience-on-mobile) article to learn how to choose between the two types of login flows.
+:::
+
+In the `onCreate` method, create a new instance of the `Auth0` class to hold user credentials:
+
+```kotlin
+// Import the parts needed by the SDK
+import com.auth0.android.Auth0
+import com.auth0.android.provider.WebAuthProvider
+
+class MainActivity : AppCompatActivity() {
+
+  private lateinit var account: Auth0
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+      super.onCreate(savedInstanceState)
+
+      // Set up the account object with the Auth0 application details
+      account = Auth0(
+          "${account.clientId}",
+          "${account.namespace}"
+      )
+  }
+}
+```
+
+:::note
+We suggest you do not hardcode the values for `clientId` and `domain` as you may need to change them in the future. Instead, use [String Resources](https://developer.android.com/guide/topics/resources/string-resource.html), such as `@string/com_auth0_domain`, to define the values. 
+:::
+
+Finally, create a `loginWithBrowser` method and use the `WebAuthProvider` class to authenticate with any connection you enabled on your application in the [Auth0 dashboard](${manage_url}/#/).
+
+```kotlin
+private fun loginWithBrowser() {
+    // Setup the WebAuthProvider, using the custom scheme and scope.
+
+    WebAuthProvider.login(account)
+        .withScheme("demo")
+        .withScope("openid profile email")
+        // Launch the authentication passing the callback where the results will be received
+        .start(this, object : Callback<Credentials, AuthenticationException> {
+            // Called when there is an authentication failure
+            override fun onFailure(exception: AuthenticationException) {
+                Snackbar.make(
+                    binding.root,
+                    "Failure: <%= "${exception.code}" %>",
+                    Snackbar.LENGTH_LONG
+                ).show()
+            }
+
+            // Called when authentication completed successfully
+            override fun onSuccess(credentials: Credentials) {
+              Snackbar.make(
+                  binding.root,
+                  "Success: <%= "${credentials.idToken}" %>",
```

Comment:
we also need to only access it if it's not null, like below 

```suggestion
                  "Success: <%= "${credentials?.idToken}" %>",
```

Reply:
Should the sample also use `idToken`? Makes sense to given that it's a login sample, but as it stands at the moment the quickstart and sample differ in this regard (the quickstart uses `idToken` but the sample uses `accessToken`).